### DISCUSSION TITLE
Where can I find more details about the metrics I am being shown when training?

### DISCUSSION THREAD

**Discussion by joawar** [explosion/spaCy#9756] [Help: Other Questions] [answered]
I am running the tagger_parser_ud project and while it's training it's showing me these metrics I have pasted below. Some of them I understand but not all, so where can I find more information about this? And how do I know how long the training will take? My config file says max_epochs = 0, max_steps = 20000. Is E the current epoch and # the number of steps? 

============================= Training pipeline =============================
ℹ Pipeline: ['tok2vec', 'tagger', 'morphologizer', 'parser']
ℹ Initial learn rate: 0.001
E    #       LOSS TOK2VEC  LOSS TAGGER  LOSS MORPH...  LOSS PARSER  TAG_ACC  POS_ACC  MORPH_ACC  DEP_UAS  DEP_LAS  SENTS_F  SCORE
---  ------  ------------  -----------  -------------  -----------  -------  -------  ---------  -------  -------  -------  ------
  0       0          0.00       148.33         149.48       291.29    20.85    23.10      25.09    17.23     6.34     1.17    0.19
  0     200       3483.71     17021.33       18480.77     29430.79    81.25    83.98      82.40    63.63    52.74    43.16    0.75

**Comment by polm** [comment, marked as answer]
> And how do I know how long the training will take? My config file says max_epochs = 0, max_steps = 20000. Is E the current epoch and # the number of steps?

This is addressed [in the FAQ](https://github.com/explosion/spaCy/discussions/8226#:~:text=What%20are%20iterations%2C%20steps%2C%20epochs....%3F). You are correct that E is the number of epochs and # the number of steps.

The other metrics are explained in [the docs](https://spacy.io/usage/training#metrics), in particular see the blue drop-down "Understanding the training output and score types".

**Reply by joawar** [reply]
Thanks, and my bad, I should have read the FAQ.
